modules/ipc/include/hephaestus/ipc/zenoh/service.h

```diff
@@ -311,8 +321,10 @@ auto callService(Session& session, const TopicConfig& topic_config, const Reques
   static constexpr auto FIFO_QUEUE_SIZE = 100;
   auto replies = session.zenoh_session.get(keyexpr, "", ::zenoh::channels::FifoChannel(FIFO_QUEUE_SIZE),
                                            std::move(options), &result);
-  throwExceptionIf<FailedZenohOperation>(result != Z_OK,
-                                         fmt::format("Failed to call service on '{}'", topic_config.name));
+  if (result != Z_OK) {
+    heph::log(heph::ERROR, "failed to call service, server error", "topic", topic_config.name);
+    return {};
```

Comment:
empty vector == failure is quite unintuitive, since this change indicates that the caller has a need to handle this error right? how about absl::StatusOr ? this would even give is the possibility to map the Zenoh errors to sth more detailed than OK or NotOK

Reply:
but the end effect is still the same, no? you make a query and you receive 0 responses. Does it matter why it failed? we had the log in any case.
I would say unless there is a clear case why we need more details on the fail I would keep it simple.
As a general rule, I prefer to do a refactor later than presume something is going to be needed eventually.